README.md

```diff
@@ -1,48 +1,207 @@
+used**, because each database works a little differently, and handling
+everything in ActiveRecord itself guarantees that we'll always get the same
+features no matter what database we're using under the hood.
+
+## What is "invalid data"?
+
+Suppose you get a new phone and you ask all of your friends for their phone
+number again. One of them tells you, "555-868-902". If you're paying attention,
+you'll probably wrinkle your nose and think, "Wait a minute. That doesn't sound
+like a real phone number."
+
+"555-868-902" is an example of **invalid data**... for a phone number. It's
+probably a valid account number for some internet service provider in Alaska,
+but there's no way to figure out what your friend's phone number is from those
+nine numbers. It's a showstopper, and even worse, it kind of looks like valid
+data if you're not looking closely.
+
+## Validations Protect the Database
+
+Invalid data is the bogeyman of web applications: it hides in your database
+until the worst possible moment, then jumps out and ruins everything by causing
+confusing errors.
+
+Imagine the phone number above being saved to the database in an application
+that makes automatic calls using the Twilio API. When your system tries to call
+this number, there will be an error because no such phone number exists, which
+means you need to have an entire branch of code dedicated to handling *just*
+that edge case.
+
+It would be much easier if you never have bad data in the first place, so you
+can focus on handling edge cases that are truly unpredictable.
+
+That's where validations come in.
+
+
+# Basic Usage
+
+For more examples of basic validation usage, see the Rails Guide for [Active
+Record Validations][ar_validations]. Take a few minutes to browse the helpers
+listed in Section 2.
+
+```ruby
+class Person < ActiveRecord::Base
+  validates :name, presence: true
+end
+
+Person.create(name: "John Doe").valid? # => true
+Person.create(name: nil).valid? # => false
+```
+
+[ar_validations]: http://guides.rubyonrails.org/active_record_validations.html
+
+`#validates()` is our Swiss Army knife for validations. It takes two arguments:
+the first is the name of the attribute we want to validate, and the second is a
+hash of options that will include the details of how to validate it.
+
+In this example, the options hash is `{ presence: true }`, which implements the
+most basic form of validation, preventing the object from being saved if its
+`name` attribute is empty.
+
+# Lifecycle Timing
+
+Before proceeding, keep the answer to this question in mind:
+
+**What is the difference between `#new()` and `#create()`?**
+
+If you've forgotten, `#new()` instantiates a new ActiveRecord model *without*
+saving it to the database, whereas `#create()` immediately attempts to save it,
+as if you had called `#new()` and then `#save()`.
+
+**Database activity triggers validation**. An ActiveRecord model instantiated
```

Comment:
Explain what database activity would mean if you used `#new`. Pretty much, validations wouldn't run until you called the method that actually hits the DB: `#save`